### PR TITLE
Improve first person camera in example

### PR DESCRIPTION
**Pull request by janhohenheim**
# Objective

- I've seen quite a few people on discord copy-paste the camera code of the first-person example and then run into problems with the pitch.
- ~~Additionally, the code is framerate-dependent.~~ it's not, see comment in PR

## Solution

- Make the code good enough to be copy-pasteable 
	- ~~Use `dt` to make the code framerate-independent~~ Add comment explaining why we don't use `dt`
	- Clamp the pitch
	- Move the camera sensitivity into a component for better configurability

## Testing

Didn't run the example again, but the code is straight from another project I have open, so I'm not worried.

### CODE REVIEW

**Comment by NiseVoid** [comment]
Multiplying mouse motion by delta time actually has the opposite effect. The mouse motion already matches the amount you moved it during the frame, so multipling it by delta time makes your mouse movement FPS-dependant, with a higher FPS meaning slower movement. However, if the input was a gamepad stick it would need to be multiplied by delta, which might be worth including in the example for that reason. Ideally we would of course upstream LWIM and have it handle those differences instead.

I do like the limit for the angle, tho the comment could be better. Not only is rotating too far not particularly useful, it also makes the other axis function like it's reversed.

File: examples/camera/first_person_view_model.rs [hunk 1]
```diff
@@ -219,18 +221,33 @@ fn spawn_text(mut commands: Commands) {
...
 
 fn move_player(
+    time: Res<Time>,
     accumulated_mouse_motion: Res<AccumulatedMouseMotion>,
     mut player: Query<&mut Transform, With<Player>>,
 ) {
     let mut transform = player.single_mut();
     let delta = accumulated_mouse_motion.delta;
+    let dt = time.delta_seconds();
+    // The factors are just arbitrary mouse sensitivity values.
```
Comment (alice-i-cecile) [inline examples/camera/first_person_view_model.rs:231]: Pull these out into constants please :)

**Comment by alice-i-cecile** [comment]
@richchurcher, if you're interested in helping out by reviewing, I'd welcome your opinions here :) Bevy uses community reviews to vet when things are ready to merge, so please leave an approval when you think this work is ready.

File: examples/camera/first_person_view_model.rs (hunk 1 above)
Comment (janhohenheim) [reply examples/camera/first_person_view_model.rs:231]: I disagree. Pulling them into `MOUSE_SENSITIVITY_X` and `MOUSE_SENSITIVITY_Y`
- makes them less nicely grouped than now, and
- is semantically wrong. `PI` is a constant, an aspect ratio is a constant. Mouse sensitivity is a value that should be a configurable variable for accessibility.

As a compromise, I could extract them into a component if you want. Or add a component mentioning that this should be configurable.

Comment (alice-i-cecile) [reply examples/camera/first_person_view_model.rs:231]: Component is nice: it makes it easy for the values to be twiddled via inspectors.

Comment (janhohenheim) [reply examples/camera/first_person_view_model.rs]: [`3e372d1` (#15109)](https://github.com/bevyengine/bevy/pull/15109/commits/3e372d1362bdd8a519dde4bcabbc4983122225b6)

**Comment by janhohenheim** [comment]
@NiseVoid thanks for pointing out that `dt` makes the mouse actually frame dependent in this case. I didn't consider that before, but it totally makes sense. Time to change all the code that works like this in my other projects :D 
Addressed all review comments.

**Review by alice-i-cecile — COMMENTED** [review]
Sticking data on the `Player` component like this feels messy, but it *does* keep the example simpler. I'd prefer a dedicated `CameraSensitivity` component.

**Comment by janhohenheim** [comment]
@alice-i-cecile yeah I was considering that as well and used the `Player` component in fear of bloat. But you're right, showing good ECS patterns is probably worth it here. Changed it :)

**Review by alice-i-cecile — APPROVED** [review]

File: examples/camera/first_person_view_model.rs [hunk 2]
```diff
@@ -220,25 +239,46 @@ fn spawn_text(mut commands: Commands) {
...
+
+        let (yaw, pitch, roll) = transform.rotation.to_euler(EulerRot::YXZ);
+        let yaw = yaw + delta_yaw;
+
+        // If the pitch was ±¹⁄₂ π, the camera would look straight up or down.
+        // When the user want to move the camera back to the horizon, which way should the camera face?
+        // The camera has no way of knowing what direction was "forward" before landing in that extreme position,
+        // so the direction picked will for all intents and purposes be arbitrary.
+        // Another issue is that for mathematical reasons, the yaw will effectively be flipped when the pitch is at the extremes.
+        // To not run into these issues, we clamp the pitch to a safe
```
Comment (chompaa) [inline examples/camera/first_person_view_model.rs:267]:
```suggestion
        // To not run into these issues, we clamp the pitch to a safe range.
```

File: examples/camera/first_person_view_model.rs [hunk 3]
```diff
@@ -220,25 +239,46 @@ fn spawn_text(mut commands: Commands) {
...
+        // as for keyboard input apply. Such an input should be multiplied by delta_time to get the intended rotation
+        // independent of the framerate.
+        let delta_yaw = -delta.x * camera_sensitivity.x;
+        let delta_pitch = -delta.y * camera_sensitivity.y;
+
+        let (yaw, pitch, roll) = transform.rotation.to_euler(EulerRot::YXZ);
+        let yaw = yaw + delta_yaw;
+
+        // If the pitch was ±¹⁄₂ π, the camera would look straight up or down.
+        // When the user want to move the camera back to the horizon, which way should the camera face?
```
Comment (chompaa) [inline examples/camera/first_person_view_model.rs:263]:
```suggestion
        // When the user wants to move the camera back to the horizon, which way should the camera face?
```

**Review by chompaa — APPROVED** [review]

File: examples/camera/first_person_view_model.rs [hunk 4]
```diff
@@ -220,25 +239,46 @@ fn spawn_text(mut commands: Commands) {
...
+
+        let (yaw, pitch, roll) = transform.rotation.to_euler(EulerRot::YXZ);
+        let yaw = yaw + delta_yaw;
+
+        // If the pitch was ±¹⁄₂ π, the camera would look straight up or down.
+        // When the user wants to move the camera back to the horizon, which way should the camera face?
+        // The camera has no way of knowing what direction was "forward" before landing in that extreme position,
+        // so the direction picked will for all intents and purposes be arbitrary.
+        // Another issue is that for mathematical reasons, the yaw will effectively be flipped when the pitch is at the extremes.
+        // To not run into these issues, we clamp the pitch to a safe range.
```
Comment (richchurcher) [inline examples/camera/first_person_view_model.rs:267]:
```suggestion
        // To not run into these issues, we clamp the pitch to a safe range. (FRAC_PI_2 is just a constant representing half the value of π).
```
Feel free to ignore this, but my first glimpse of this value gave me a bit of a WTF moment :laughing:

**Review by richchurcher — APPROVED** [review]
All looks very familiar!

File: examples/camera/first_person_view_model.rs (hunk 4 above)
Comment (janhohenheim) [reply examples/camera/first_person_view_model.rs]: Good point :) I changed the wording a bit: [`2bc51d1` (#15109)](https://github.com/bevyengine/bevy/pull/15109/commits/2bc51d11968cd78ead6b4c82119ed18ab8f05e10)

**Pull request merged**